src/mca/pdl/pdlopen/pdl_pdlopen_module.c

```diff
@@ -175,7 +175,26 @@ static int pdlopen_foreachfile(const char *search_path,
 
         dp = opendir(dirs[i]);
         if (NULL == dp) {
-            ret = PMIX_ERR_IN_ERRNO;
+            if (0 == i) {
```

Comment:
I chatted with Nathan and he sees no issue with continuing the search beyond the first failure, so I've added that here and fixed the bug being seen in the OMPI version